mitmproxy/addons/command_history.py

```diff
@@ -0,0 +1,113 @@
+import collections
+import os
+import typing
+import atexit
+
+from mitmproxy import command
+from mitmproxy import ctx
+
+
+class CommandHistory:
+    def __init__(self, size: int = 300) -> None:
+        self.saved_commands: typing.Deque[str] = collections.deque(maxlen=size)
+
+        self.filtered_commands: typing.Deque[str] = collections.deque()
+        self.current_index: int = -1
+        self.filter_str: str = ''
+
+        _command_history_dir = os.path.expanduser(ctx.options.confdir)
+        if not os.path.exists(_command_history_dir):
+            os.makedirs(_command_history_dir)
```

Comment:
What happens if you do all this in `configure`? Is the directory available then?

Reply:
Not sure... I will test and see if it works

Reply:
Seems to work fine with `configure`. Let me know what you think of the changes.